### ISSUE TITLE
"The clean action has not yet been ported"

### ISSUE THREAD

**Issue by JohannesMP** [premake/premake-core#278]
$ premake5 clean
    Running action 'clean'...
    ** The clean action has not yet been ported

-----

- Premake 5.0.0.alpha4
- macosx 10.10.5

**Comment by nickclark2016** [comment]
I don't believe there are any blockers on the issue. It's just been a lower priority feature to port for us. We'd happily accept any PRs that want to port the feature. The tricky part for implementing "clean" would be needing to go through all of the possible actions, determining their output configuration files, then only removing those.

As for if people use it, I know I do not. I've never really come across a need for it.

**Comment by Jarod42** [comment]
I have investigated on the task, and there is API problem for me:
- `location` might be given as `location 'solution/' + _ACTION` which cannot be correctly retrieved (`_ACTION` would be `"clean"` for clean action and not `"vs2022"` or `"gmake"`.
`location 'solution/%{_ACTION}` might mitigate above the issue, but I don't succeed to have proper `location` neither (I think baking happens too early) but should be doable.

Having `_OPTIONS` for clean-up seems better IMO: `premake vs2022 --remove-solution`.

**Comment by JohannesMP** [comment]
This may be too philosophical, but if it doesn't do anything, and it's not used, and it's too much work to update it given priority of other tasks... just kill it. Remove it from the API.

If the feature is ever needed, add it as a new feature then. If the old code is needed them, that is what we use version control for.


As it stands this just reads as a forever `#todo` comment that is visible to the end user.
